src/Faithlife.WebRequests/WebServiceRequestBase.cs

```diff
@@ -289,19 +264,19 @@ private HttpContent GetRequestContent(HttpRequestMessage webRequest)
 			return requestContent;
 		}
 
-		private async Task<TResponse> HandleResponseAsync(HttpRequestMessage webRequest, HttpResponseMessage webResponse, IWorkState workState)
+		private async Task<TResponse> HandleResponseAsync(HttpRequestMessage webRequest, HttpResponseMessage webResponse, CancellationToken cancellationToken)
 		{
 			// handle web response
-			WebServiceResponseHandlerInfo<TResponse> info = new WebServiceResponseHandlerInfo<TResponse>(webResponse, workState);
+			WebServiceResponseHandlerInfo<TResponse> info = new WebServiceResponseHandlerInfo<TResponse>(webResponse, cancellationToken);
 			try
 			{
 				if (!await HandleResponseCoreAsync(info).ConfigureAwait(false))
 				{
-					if (workState.Canceled)
+					if (cancellationToken.IsCancellationRequested)
```

Comment:
I hate exceptions for flow control as much as (probably more than) the next guy, but should we be doing `cancellationToken.ThrowIfCancellationRequested()` here?

Reply:
Agreed. Callers would expect cancellation to throw, not return null.